apps/webcomponents/src/app/components/gn-figure-datasets/gn-figure-datasets.component.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Injector,
+  ViewEncapsulation,
+} from '@angular/core'
+import { BaseComponent } from '../base.component'
+import { RecordsService } from '@geonetwork-ui/feature/catalog'
+import { startWith } from 'rxjs/operators'
+import { Observable } from 'rxjs'
+import { SearchFacade } from '@geonetwork-ui/feature/search'
+
+@Component({
+  selector: 'wc-gn-figure-datasets',
+  templateUrl: './gn-figure-datasets.component.html',
+  styleUrls: ['./gn-figure-datasets.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.ShadowDom,
+  providers: [SearchFacade],
+})
+export class GnFigureDatasetsComponent extends BaseComponent {
+  catalogRecords: RecordsService
+  recordsCount$: Observable<string | number>
+
+  constructor(injector: Injector, private changeDetector: ChangeDetectorRef) {
+    super(injector)
+    this.catalogRecords = injector.get(RecordsService)
+    this.recordsCount$ = this.catalogRecords.recordsCount$.pipe(startWith('-'))
```

Comment:
I would add a` catchError(()=> of('-'))` to stay consistent with the way you built it. 
I like the way that you start with an '-'.
Since you put : `catchError(() => of(0))` on the `recordsCount$` in the service, in case of error it will display to the user 0 wich is wrong (could be 14 or 392).

What do you think about this ?